File: CODIGOS5/173.swift
func determineChordName(notes: [Int]) -> String {
    var steps = [Int]()
    for i in 1..<notes.count {
        let diff = (notes[i] - notes[i - 1] + 12) % 12
        steps.append(diff)
    }

    if steps.contains(3) && steps.contains(4) {
        return "other"
    } else if steps.contains(4) {
        return getChordName(rootNote: notes[0], isMajor: true)
    } else if steps.contains(3) {
        return getChordName(rootNote: notes[0], isMajor: false)
    } else {
        return "other"
    }
}

func getChordName(rootNote: Int, isMajor: Bool) -> String {
    let noteNames = ["C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B"]
    let rootNoteName = noteNames[rootNote % 12]
    if isMajor {
        return "\(rootNoteName)-major"
    } else {
        return "\(rootNoteName)-minor"
    }
}

func main() {
    guard let numTestCasesStr = readLine(),
          let numTestCases = Int(numTestCasesStr) else {
        return
    }

    for _ in 1...numTestCases {
        guard let inputNotesStr = readLine() else {
            return
        }
        let inputNotes = inputNotesStr.split(separator: " ").compactMap { Int($0) }
        let chordName = determineChordName(notes: inputNotes)
        print("\(chordName) ", terminator: "")
    }
}

main()

File: CODIGOS5/173.rs
use std::io;

fn determine_chord_name(notes: &[i32]) -> String {
    let mut steps = Vec::new();
    for i in 1..notes.len() {
        let diff = (notes[i] - notes[i - 1] + 12) % 12;
        steps.push(diff);
    }

    if steps.contains(&3) && steps.contains(&4) {
        return "other".to_string();
    } else if steps.contains(&4) {
        return get_chord_name(notes[0], true);
    } else if steps.contains(&3) {
        return get_chord_name(notes[0], false);
    } else {
        return "other".to_string();
    }
}

fn get_chord_name(root_note: i32, is_major: bool) -> String {
    let note_names = vec![
        "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B",
    ];
    let root_note_name = note_names[(root_note % 12) as usize];
    if is_major {
        return format!("{}-major", root_note_name);
    }
    format!("{}-minor", root_note_name)
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let num_test_cases: i32 = input.trim().parse().unwrap();

    for _ in 0..num_test_cases {
        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let notes: Vec<i32> = input
            .split_whitespace()
            .map(|x| x.parse().unwrap())
            .collect();

        let chord_name = determine_chord_name(&notes);
        print!("{} ", chord_name);
    }
}
